src/Database/LSMTree/Internal/RunReader.hs

```diff
@@ -187,10 +209,28 @@ next fs hbio reader@RunReader {..} = do
   Utilities
 -------------------------------------------------------------------------------}
 
+-- | Returns 'Nothing' on EOF.
+readDiskPageAt :: HasFS IO h -> Index.PageNo -> FS.Handle h -> IO (Maybe RawPage)
+readDiskPageAt fs pageNo h = do
+    mba <- newPinnedByteArray pageSize
+    -- TODO: make sure no other exception type can be thrown
```

Comment:
Others will just propagate. Is that ok?

Reply:
After having some weeks to think about it, I don't think we can properly handle other exceptions at this call site.

Reply:
@dcoutts I have added a paraphrased version of your commentary as a comment to `RunReader.new`.